### ISSUE TITLE
Question about backtype.storm and com.twitter.heron.api

### ISSUE THREAD

**Issue by xcf1992** [twitter/heron#1243]
Hi @maosongfu,

I have created a simple topology, everything goes fine if I use topologyBuilder, spout, bolt and tuple from backtype.storm.

However, when I change to use these objects from com.twitter.heron.api, one of bolts keep getting:
[2016-08-10 14:10:37 -0700] com.twitter.heron.network.StreamManagerClient INFO:  Stop reading due to not yet connected to Stream Manager.  
[2016-08-10 14:10:37 -0700] com.twitter.heron.network.StreamManagerClient INFO:  Stop writing due to not yet connected to Stream Manager.  
[2016-08-10 14:10:37 -0700] com.twitter.heron.network.StreamManagerClient INFO:  Stop reading due to not yet connected to Stream Manager.  
[2016-08-10 14:10:37 -0700] com.twitter.heron.network.StreamManagerClient INFO:  Stop writing due to not yet connected to Stream Manager.  
[2016-08-10 14:10:37 -0700] com.twitter.heron.network.StreamManagerClient INFO:  Stop reading due to not yet connected to Stream Manager.  
[2016-08-10 14:10:38 -0700] com.twitter.heron.network.StreamManagerClient INFO:  Stop writing due to not yet connected to Stream Manager.  
[2016-08-10 14:10:38 -0700] com.twitter.heron.network.StreamManagerClient INFO:  Stop reading due to not yet connected to Stream Manager.  
[2016-08-10 14:10:38 -0700] com.twitter.heron.network.StreamManagerClient INFO:  Stop writing due to not yet connected to Stream Manager.  
[2016-08-10 14:10:38 -0700] com.twitter.heron.network.StreamManagerClient INFO:  Stop reading due to not yet connected to Stream Manager.  
[2016-08-10 14:10:38 -0700] com.twitter.heron.common.network.HeronClient INFO:  Connecting to endpoint: /127.0.0.1:56824  
[2016-08-10 14:10:38 -0700] com.twitter.heron.common.network.HeronClient SEVERE:  Failed to FinishConnect to endpoint: /127.0.0.1:56824 
java.net.ConnectException: Connection refused
	at sun.nio.ch.SocketChannelImpl.checkConnect(Native Method)
	at sun.nio.ch.SocketChannelImpl.finishConnect(SocketChannelImpl.java:717)
	at com.twitter.heron.common.network.HeronClient.handleConnect(HeronClient.java:244)
	at com.twitter.heron.common.basics.NIOLooper.handleSelectedKeys(NIOLooper.java:115)
	at com.twitter.heron.common.basics.NIOLooper.access$000(NIOLooper.java:32)
	at com.twitter.heron.common.basics.NIOLooper$1.run(NIOLooper.java:45)
	at com.twitter.heron.common.basics.WakeableLooper.executeTasksOnWakeup(WakeableLooper.java:142)
	at com.twitter.heron.common.basics.WakeableLooper.runOnce(WakeableLooper.java:74)
	at com.twitter.heron.common.basics.WakeableLooper.loop(WakeableLooper.java:64)
	at com.twitter.heron.instance.Gateway.run(Gateway.java:155)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at java.lang.Thread.run(Thread.java:745)

I am wondering any special reason cause this different? 

Thanks

**Comment by maosongfu** [comment]
I don't think it is related to heron.api; could u share the log-files in your sandbox?

BTW,  heron.api is not encouraged to use for now since it is not stable and would be changed quite a bit in near future.

**Comment by xcf1992** [comment]
Ok, I will keep using backtype.storm, but is there performance different between backtype and heron.api.

Besides, which specific file you want to take a look at?

**Comment by maosongfu** [comment]
There is no performance difference between backtype and heron.api.

I want the logs mentioned in here: http://twitter.github.io/heron/docs/getting-started-troubleshooting/
~/.herondata/topologies/{cluster}/{role}/{TopologyName}/heron-executor.stdout
~/.herondata/topologies/{cluster}/{role}/{TopologyName}/log-files/

**Comment by kramasamy** [comment]
@xcf1992 - the com.twitter.heron.api is an internal API and you are discouraged to use it. You should use either backtype.storm.* or org.apache.storm.*

**Comment by xcf1992** [comment]
I think this these files files may be the most important:

[heron-executor.stdout.txt](https://github.com/twitter/heron/files/412318/heron-executor.stdout.txt)

[container_1_g_6.log.0.txt](https://github.com/twitter/heron/files/412321/container_1_g_6.log.0.txt)
[heron-stmgr-stmgr-1.log.INFO.20160810-160954.14282.txt](https://github.com/twitter/heron/files/412327/heron-stmgr-stmgr-1.log.INFO.20160810-160954.14282.txt)

Cause I have to remove any info about the service or company, let me know if any other file may also neededed.

Thanks

**Comment by kramasamy** [comment]
@maosongfu - can you please take a look at this?

**Comment by maosongfu** [comment]
According to the container_1_g_6.log.0, it is a known issue:
```
[2016-08-10 16:10:01 -0700] com.twitter.heron.instance.HeronInstance SEVERE:  Exception caught in thread: SlaveThread with id: 13 
java.lang.NoClassDefFoundError: com/esotericsoftware/reflectasm/ConstructorAccess
	at com.esotericsoftware.kryo.Kryo$DefaultInstantiatorStrategy.newInstantiatorOf(Kryo.java:1233)
	at com.esotericsoftware.kryo.Kryo.newInstantiator(Kryo.java:1078)
	at com.esotericsoftware.kryo.Kryo.newInstance(Kryo.java:1087)
	at com.esotericsoftware.kryo.serializers.CollectionSerializer.create(CollectionSerializer.java:107)
	at com.esotericsoftware.kryo.serializers.CollectionSerializer.read(CollectionSerializer.java:111)
	at com.esotericsoftware.kryo.serializers.CollectionSerializer.read(CollectionSerializer.java:40)
	at com.esotericsoftware.kryo.Kryo.readClassAndObject(Kryo.java:793)
	at com.twitter.heron.api.serializer.KryoSerializer.deserialize(KryoSerializer.java:49)
	at com.twitter.heron.instance.bolt.BoltInstance.handleDataTuple(BoltInstance.java:170)
	at com.twitter.heron.instance.bolt.BoltInstance.readTuplesAndExecute(BoltInstance.java:217)
	at com.twitter.heron.instance.bolt.BoltInstance$1.run(BoltInstance.java:142)
	at com.twitter.heron.common.basics.WakeableLooper.executeTasksOnWakeup(WakeableLooper.java:142)
	at com.twitter.heron.common.basics.WakeableLooper.runOnce(WakeableLooper.java:74)
	at com.twitter.heron.common.basics.WakeableLooper.loop(WakeableLooper.java:64)
	at com.twitter.heron.instance.Slave.run(Slave.java:169)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at java.lang.Thread.run(Thread.java:745)
Caused by: java.lang.ClassNotFoundException: com.esotericsoftware.reflectasm.ConstructorAccess
	at java.net.URLClassLoader.findClass(URLClassLoader.java:381)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:424)
	at sun.misc.Launcher$AppClassLoader.loadClass(Launcher.java:331)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:357)
	... 18 more
```

which is reported by @ajorgensen in https://github.com/twitter/heron/issues/1165. And it is fixed in latest master branch and release 0.14.2.
@xcf1992 Can u give it a try on 0.14.2? Thanks.

**Comment by xcf1992** [comment]
Thanks a lot. I will try 0.14.2 to see if this get fixed.
